File: Algorithm_114_Flatten Binary Tree to Linked List.cpp
#include <iostream>
using namespace std;
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Solution
{
    public:
        TreeNode *prev;
        void flatten(TreeNode *root)
        {
            if (!root) return;
            flatten(root->right);
            flatten(root->left);
            root->right = prev;
            root->left = NULL;
            prev = root;
            return;
        }
};

File: Algorithm_114_Flatten Binary Tree to Linked List.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def flatten(self, root: TreeNode) -> None:
        self.prev = None
        self.helper(root)
        
    def helper(self, root: TreeNode) -> None:
        if not root:
            return
        self.helper(root.right)
        self.helper(root.left)
        root.right = self.prev
        root.left = None
        self.prev = root